thermal.c

```diff
@@ -14,43 +14,333 @@
 
 cpumask_t thermal_banned_cpus;
 
+#define THERMAL_GENL_FAMILY_NAME	"thermal"
+#define THERMAL_GENL_EVENT_GROUP_NAME	"event"
+
+struct family_data {
+	const char *group;
+	int id;
+};
+
+static struct nl_sock *sock;
+static struct nl_cb *callback;
+
 static gboolean prepare_netlink(void)
 {
-	gboolean error = TRUE;
+	gboolean error = FALSE;
+	int rc;
+
+	sock = nl_socket_alloc();
+	if (!sock) {
+		log(TO_ALL, LOG_ERR, "thermal: socket allocation failed.\n");
+		goto err_out;
+	}
 
-	log(TO_ALL, LOG_ERR, "thermal: not yet implement to alloc memory for netlink.\n");
+	rc = genl_connect(sock);
+	if (rc) {
+		log(TO_ALL, LOG_ERR, "thermal: socket bind failed.\n");
+		goto err_out;
+	}
+
+	callback = nl_cb_alloc(NL_CB_DEFAULT);
+	if (!callback) {
+		log(TO_ALL, LOG_ERR, "thermal: callback allocation failed.\n");
+		goto err_out;
+	}
+
+	goto out;
+err_out:
```

Comment:
Just set error=false here and fall through rather than having an extra goto

Reply:
?

Reply:
Sorry, I missed this. I don't recall why..

Yeah, I now think these whole gotos are not necessary. Simply return TRUE on every failure. And the fall-through returns FALSE. Then explain the implication of this boolean value at the function comment.

Will set the resolve button along with my revision.